CMakeLists.txt

```diff
@@ -52,6 +52,21 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror")
 
+# Determine if CXX Compiler is hcc, hip-clang or other
```

Comment:
It would be better to move it to `cmake/VerifyCompiler.cmake` after line 37, so this code it not run if `HIP_PLATFORM` is `nvcc` and only hipCUB will be built.

Reply:
I think I made that comment in the first "big" PR, but I forgot about it here when I did the first review of this PR.

Reply:
Thanks, that's a good point